sleap_io/io/slp.py

```diff
@@ -1055,6 +1060,525 @@ def write_lfs(labels_path: str, labels: Labels):
+        "translation": camera.tvec.tolist(),
+    }
+    camera_dict.update(camera.metadata)
+
+    return camera_dict
+
+
+def camera_group_to_dict(camera_group) -> dict:
+    """Convert `CameraGroup` to dictionary.
+
+    Returns:
+        Dictionary containing camera group information with the following keys:
+            cam_n: Camera dictionary containing information for camera at index "n"
+                with the following keys:
+                name: Camera name.
+                size: Image size (height, width) of camera in pixels of size (2,)
+                    and type int.
+                matrix: Intrinsic camera matrix of size (3, 3) and type float64.
+                distortions: Radial-tangential distortion coefficients
+                    [k_1, k_2, p_1, p_2, k_3] of size (5,) and type float64.
+                rotation: Rotation vector in unnormalized axis-angle representation
+                    of size (3,) and type float64.
+                translation: Translation vector of size (3,) and type float64.
+            metadata: Dictionary of metadata.
+    """
+    calibration_dict = {}
+    for cam_idx, camera in enumerate(camera_group.cameras):
+        camera_dict = camera_to_dict(camera)
+        calibration_dict[f"cam_{cam_idx}"] = camera_dict
+
+    calibration_dict["metadata"] = camera_group.metadata.copy()
+
+    return calibration_dict
+
+
+def session_to_dict(
+    session,
+    video_to_idx: dict[Video, int],
+    labeled_frame_to_idx: dict[LabeledFrame, int],
+) -> dict:
+    """Unstructure `RecordingSession` to an invertible dictionary.
+
+    Args:
+        video_to_idx: Dictionary of `Video` to index in `Labels.videos`.
+        labeled_frame_to_idx: Dictionary of `LabeledFrame` to index in
+            `Labels.labeled_frames`.
+
+    Returns:
+        Dictionary of "calibration" and "camcorder_to_video_idx_map" needed to
+        restructure a `RecordingSession`.
+    """
+    # Unstructure `CameraCluster` and `metadata`
+    calibration_dict = camera_group_to_dict(session.camera_group)
+
+    # Store camera-to-video indices map where key is camera index
+    # and value is video index from `Labels.videos`
+    camera_to_video_idx_map = {}
+    for cam_idx, camera in enumerate(session.camera_group.cameras):
+        # Skip if Camera is not linked to any Video
+
+        # TODO(LM): Do not call private attributes outside of class
+        if camera not in session._video_by_camera:
+            continue
+
+        # Get video index from `Labels.videos`
+
+        # TODO(LM): Do not call private attributes outside of class
+        video = session._video_by_camera[camera]
+        video_idx = video_to_idx.get(video, None)
+
+        if video_idx is not None:
+            camera_to_video_idx_map[cam_idx] = video_idx
+        else:
+            print(
+                f"Video {video} not found in `Labels.videos`. "
+                "Not saving to `RecordingSession` serialization."
+            )
```

Comment:
> When does this happen? Wouldn't it be better to throw an Exception and catch it upstream if needed?

_Originally posted by @talmo in https://github.com/talmolab/sleap-io/pull/151#discussion_r1956956588_

Reply:
>I added many of these sorts of checks initially to give me useful feedback for what may have gone wrong had I messed something up. The case I could see this happening is if we remove a `Video` from `Labels.videos` but do not update the linking in `RecordingSession._video_by_camera` (in which case we don't want that deleted `Video` serialized).
>
>I did not raise an error here because, while I want to give some sort of warning, I don't want the entire operation to fail due to one missing video.

_Originally posted by @roomrys in https://github.com/talmolab/sleap-io/pull/151#discussion_r1964558997_